### PR TITLE
Dimensionality reduction

### PR DESCRIPTION
**Pull request by gathurian**

### CODE REVIEW

File: FS19/ML/Zusammenfassung_ML.tex [hunk 1]
```diff
@@ -764,105 +764,105 @@ \section{Gradient Descent}
...
+        6(3x + 2y) \\
+        4(3x + 2y)
+    \end{bmatrix} \\
+        \nabla f(1,2) & = \begin{bmatrix}
+            42 \\
+            28
+        \end{bmatrix}
+        \end{align*}
+
+        \subsection{Properties of the gradient}
```
Comment (KilnOfTheSecondFlame) [inline FS19/ML/Zusammenfassung_ML.tex]: The auto-formatter strikes again?

File: FS19/ML/Zusammenfassung_ML.tex [hunk 2]
```diff
@@ -2370,6 +2373,215 @@ \subsection{Evaluation}
...
+
+    Generally, the following holds true:
+
+    \begin{quote}
+        \blockquote{The smaller the error, the larger the variance, the more informative the projection}
+    \end{quote}
+
+    As can be seen in fig. \ref{fig:retained_information}, the further away the vector strays from the variance term, the larger the projection error grows. This in turn falsifies the information. 
+
+    Therefore: The best projection is the one that minimizes information loss or equivalently that maximizes the variance.
```
Comment (KilnOfTheSecondFlame) [inline FS19/ML/Zusammenfassung_ML.tex]: minimises and maximises
I wrote British the whole time =/

File: FS19/ML/Zusammenfassung_ML.tex [hunk 3]
```diff
@@ -2370,6 +2373,215 @@ \subsection{Evaluation}
...
+
+\subsection{Data Redundancy}
+
+As said in the previous page: With almost any dimension reduction, some information will get lost.
+
+However, sometimes this does not hold true. This is the case if some of the data is \textit{redundant}
+
+\vspace{10px}
+
+Take for example a \textbf{mean-centered feature} that always takes the same value (so, the mean). This feature dow not give you any valuable information and can therefore safely be dropped.
```
Comment (KilnOfTheSecondFlame) [inline FS19/ML/Zusammenfassung_ML.tex]: This feature _does_ not give you [...]

File: FS19/ML/Zusammenfassung_ML.tex [hunk 4]
```diff
@@ -2370,6 +2373,215 @@ \subsection{Evaluation}
...
+
+Another example are \textbf{several features that essentially say the same}, say the height of a test person in centimeters and in inches. One of the two can be dropped. In the unlikely case that both measurements need to be used, one can easily be calculated.
```
Comment (KilnOfTheSecondFlame) [inline FS19/ML/Zusammenfassung_ML.tex]: [...], one can easily be calculated _from the other_.

File: FS19/ML/Zusammenfassung_ML.tex [hunk 5]
```diff
@@ -2370,6 +2373,215 @@ \subsection{Evaluation}
...
+\end{align*}
+
+\noindent $P$ essentially just rotates and stretches $X$ to transform it into $Y$. Therefore, $P$ performs \textbf{base transformation} on $X$.
+
+\noindent The rows of $P$ are the new base vectors of $Y$, also called \textbf{principal components}
+
+\subsection{Eigenvectors and Eigenvalues}
+
+\begin{description}
+    \item[Eigenvector: ] A vector that does only stretch or rotate (but noth both) a matrix $A$ when multiplied with the matrix
```
Comment (KilnOfTheSecondFlame) [inline FS19/ML/Zusammenfassung_ML.tex]: An Eigenvector is a vector that is a only a multiple of itself (i.e. stretches) when multiplied with the matrix A, no rotation!

File: FS19/ML/Zusammenfassung_ML.tex [hunk 6]
```diff
@@ -2370,6 +2373,215 @@ \subsection{Evaluation}
...
+    \item[Eigenvalue: ] A factor of the eigenvector
```
Comment (KilnOfTheSecondFlame) [inline FS19/ML/Zusammenfassung_ML.tex]: The scalar value by which the vector is stretched when multiplied with the matrix
The equation looks as follows: Av = \lambda v

File: FS19/ML/Zusammenfassung_ML.tex [hunk 7]
```diff
@@ -2370,6 +2373,215 @@ \subsection{Evaluation}
...
+\end{description}
+
+\noindent A vector $ v \neq 0$ is called an eigenvector if it satisfies $Av = \lambda v$. $\lambda$ is called the eigenvalue of the eigenvector $v$.
+
+Eigenvectors (and therefore eigenvalues) only exist for \textbf{squared matrices}.o
```
Comment (KilnOfTheSecondFlame) [inline FS19/ML/Zusammenfassung_ML.tex]: Lose the o

File: FS19/ML/Zusammenfassung_ML.tex [hunk 8]
```diff
@@ -2370,6 +2373,215 @@ \subsection{Evaluation}
...
+
+In practice:
+
+\begin{enumerate}
+    \item Each eigenvector (row) in $P = E^T$ matches to one eigenvalue in $S_Y$
+    \item sort the eigenvalues in $S_Y$ in decreasing order and rearrange the rows in $P$
+    \item Eliminate features (rows) from the lowest to the highest eigenvalue
+\end{enumerate}
+    
+
```
Comment (KilnOfTheSecondFlame) [inline FS19/ML/Zusammenfassung_ML.tex]: So many newlines: necessary?

**Review by KilnOfTheSecondFlame — CHANGES_REQUESTED** [review]

File: FS19/ML/Zusammenfassung_ML.tex (hunk 1 above)
Comment (gathurian) [reply FS19/ML/Zusammenfassung_ML.tex]: Can't handle free software as it seems...

File: FS19/ML/Zusammenfassung_ML.tex [hunk 9]
```diff
@@ -2468,18 +2468,18 @@ \subsection{Strategies for Dimensionality Reduction} \label{sec:strategies}
 
 \begin{enumerate}
     \item Eliminate redundancy \\
-        Fully redundant features can be dropped without any information loss \\
-        Combine features in order to remove all covariance in the data
+          Fully redundant features can be dropped without any information loss \\
+          Combine features in order to remove all covariance in the data
     \item Destroy as little information as possible \\
-        Mimize the information loss during the projection by maximizing the retained variance in the projected data.
+          Mimize the information loss during the projection by maximizing the retained variance in the projected data.
```
Comment (KilnOfTheSecondFlame) [inline FS19/ML/Zusammenfassung_ML.tex:2474]: Minimize / Minimise, sorry did not see that one

File: FS19/ML/Zusammenfassung_ML.tex [hunk 10]
```diff
@@ -2508,13 +2508,14 @@ \subsection{Base Transformation on Data Matrices}
 \subsection{Eigenvectors and Eigenvalues}
 
 \begin{description}
-    \item[Eigenvector: ] A vector that does only stretch or rotate (but noth both) a matrix $A$ when multiplied with the matrix
-    \item[Eigenvalue: ] A factor of the eigenvector
+    \item[Eigenvector: ] A vector that is a only a multiple of itself (i.e. stretches) when multiplied with the matrix $A$, no rotation!
```
Comment (KilnOfTheSecondFlame) [inline FS19/ML/Zusammenfassung_ML.tex:2511]: Lose the 'no rotation' - that was purely for your information

**Review by KilnOfTheSecondFlame — CHANGES_REQUESTED** [review]

File: FS19/ML/Zusammenfassung_ML.tex (hunk 10 above)
Comment (gathurian) [reply FS19/ML/Zusammenfassung_ML.tex:2511]: I think it's useful to emphasise that it does purely stretch the vector

**Review by KilnOfTheSecondFlame — APPROVED** [review]

**Pull request merged**